xdg-ninja.sh

```diff
@@ -5,91 +5,172 @@ has_command() {
+    ANSI_BOLD_CLEAR="\033[;1;0m"
+    ANSI_BOLD_CYAN="\033[;1;37m"
+    ANSI_BOLD_GREEN="\033[;1;32m"
+    ANSI_BOLD_ITALIC="\033[;1;3m"
+    ANSI_BOLD_RED="\033[;1;31m"
+    ANSI_BOLD_WHITE_BACKGROUND_PURPLE="\033[;1;37;45m"
+    ANSI_BOLD_YELLOW="\033[;1;33m"
+}
+
+help() {
+    set_colors
+    HELPSTRING="""\
+
+
+        ${ANSI_BOLD_WHITE_BACKGROUND_PURPLE}xdg-ninja${ANSI_CLEAR}
 
+        ${ANSI_BOLD_ITALIC}Check your \$HOME for unwanted files.${ANSI_BOLD_CLEAR}
 
-    \033[37;45;1mxdg-ninja\033[0m
+        ────────────────────────────────────
 
-    \033[1;3mCheck your \$HOME for unwanted files.\033[1;0m
+        ${ANSI_ITALIC}--help${ANSI_CLEAR}               ${ANSI_BOLD}This help menu${ANSI_CLEAR}
+        ${ANSI_ITALIC}-h${ANSI_CLEAR}
 
-    ────────────────────────────────────
+        ${ANSI_ITALIC}--skip-ok${ANSI_CLEAR}            ${ANSI_BOLD}Don't display anything for files that do not exist (default)${ANSI_CLEAR}
+        ${ANSI_ITALIC}--skip-warn${ANSI_CLEAR}          ${ANSI_BOLD}Don't display anything for files that cannot be fixed.${ANSI_CLEAR}
+        ${ANSI_ITALIC}--no-skip-ok${ANSI_CLEAR}         ${ANSI_BOLD}Display messages for all files checked${ANSI_CLEAR}
+        ${ANSI_ITALIC}--no-skip-warn${ANSI_CLEAR}       ${ANSI_BOLD}Don't display anything for files that cannot be fixed.${ANSI_CLEAR}
+        ${ANSI_ITALIC}--color=WHEN${ANSI_CLEAR}         ${ANSI_BOLD}Color the output always, never, or auto (default)${ANSI_CLEAR}
+        ${ANSI_ITALIC}--decoder=DECODER${ANSI_CLEAR}    ${ANSI_BOLD}Manually set the decoder used for markdown.${ANSI_CLEAR}
+        ${ANSI_ITALIC}--output-style=STYLE${ANSI_CLEAR} ${ANSI_BOLD}Style of ouptut, either normal (defualt) or json${ANSI_CLEAR}
 
-    \033[3m--help\033[0m              \033[1mThis help menu\033[0m
-    \033[3m-h\033[0m
+        ${ANSI_ITALIC}--quiet${ANSI_CLEAR}              ${ANSI_BOLD}Causes program to run quietely${ANSI_CLEAR}
+        ${ANSI_ITALIC}-q${ANSI_CLEAR}
 
-    \033[3m--no-skip-ok\033[0m        \033[1mDisplay messages for all files checked (verbose)\033[0m
-    \033[3m-v\033[0m
+        ${ANSI_ITALIC}--loud${ANSI_CLEAR}               ${ANSI_BOLD}Causes program to not run quietely${ANSI_CLEAR}
 
-    \033[3m--skip-ok\033[0m           \033[1mDon't display anything for files that do not exist (default)\033[0m
+        ${ANSI_ITALIC}--verbose${ANSI_CLEAR}            ${ANSI_BOLD}Combines --no-skip-warn and --no-skip-ok${ANSI_CLEAR}
+        ${ANSI_ITALIC}-v${ANSI_CLEAR}
 
 """
+printf "%b" "$HELPSTRING"
+exit
+}
 
 SKIP_OK=true
+SKIP_WARN=false
+COLOR=auto
+OUTPUT_STYLE=normal
+EXIT_STATUS=0
+HELP=false
+alias loudprintf="printf"
 for i in "$@"; do
-    if [ "$i" = "--help" ] || [ "$i" = "-h" ]; then
-        printf "%b" "$HELPSTRING"
-        exit
-    elif [ "$i" = "--skip-ok" ]; then
-        SKIP_OK=true
-    elif [ "$i" = "--no-skip-ok" ]; then
-        SKIP_OK=false
-    elif [ "$i" = "-v" ]; then
-        SKIP_OK=false
-    fi
+    case $i in
```

Comment:
You still haven't implemented proper argument parsing

Reply:
I did... It was rejected.

Reply:
Oh, ok